### PR TITLE
Task 3 components

### PR DESCRIPTION
**Pull request by mill9r**
Task 3 Components
View example: 
image

1)Markup is done with React Components **(Done)**
2)Use<ErrorBoundary>component for catching and displaying errors **(Done)**
image
3)Use smart/dumb components approach **(Done)**
4)Decomposition **(Done)**

How to start:
Run `npm start` in the root folder

### CODE REVIEW

**Comment by dpryshchep** [comment]
I would strongly recommend to place components in scope of separate folder (`components`) instead of placing them directly into `scr`
```
-- components
   -- card
   -- common
   -- etc.
```

**Comment by dpryshchep** [comment]
please add `.gitignore` file (build, node_modules, etc)
add package-lock.json or yarn-lock.json (depends on what you exactly use)

**Comment by dpryshchep** [comment]
PR Template example:

### What This Change Does
* place you notes here

### Things Reviewers Should Pay Attention To
* place you notes here

### Screenshots 
* visual proof of changes (if they are exists)

### Useful Links
[Task description](place you link here)
[Deployed demo](place you link here)
...etc

File: src/card/MovieCard.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import Image from "react-bootstrap/Image";
+import Nav from "react-bootstrap/Nav";
+
+
+export const MovieCard = (props) => {
+
+    const addDefaultSrc = (e) => {
+        e.target.src = "img/fallback.png";
+        e.target.onerror = null;
+
```
Comment (dpryshchep) [inline src/card/MovieCard.js:14]: redundant line

File: src/common/ButtonGroup.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    let buttons = props.buttons.map(button => {
+            return (<ToggleButton
+                size="small"
+                key={button.name}
+                className={` ${props.white ? "text-white":""}  ${button.selected ? "selected-button" : ""}`}
+                value={button.name}
+                selected={button.selected}>
+                {button.name}
+            </ToggleButton>)
```
Comment (dpryshchep) [inline src/common/ButtonGroup.js:16]: Formatting please
```
return (
  <ToggleButton
     size="small"
     key={button.name}
     className={` ${props.white ? "text-white":""}  ${button.selected ? "selected-button" : ""}`}
     value={button.name}
     selected={button.selected}>
     {button.name}
   </ToggleButton>
)
```

File: src/details/DetailsContainer.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        fetchData();
+    },[data.id]);
+
+    return (
+        <>
+        {data.id && <Details {...data}/>}
+        {moviesByGenres && moviesByGenres.length > 0 &&
+        <ControlBar genres={data.genres} ></ControlBar>}
+        <Content response={moviesByGenres}></Content>
+        </>
```
Comment (dpryshchep) [inline src/details/DetailsContainer.js:30]: Formatting and components definition without children component
```
<>
  {data.id && <Details {...data}/>}
  {moviesByGenres && moviesByGenres.length > 0 &&
        <ControlBar genres={data.genres} />
  }
  <Content response={moviesByGenres} />
</>
```

File: src/header/Header.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                    <Typography gutterBottom
+                                color="secondary"
+                                className={classes.title}
+                                variant="h5"><strong>netflix</strong>roulette</Typography>
+                </Grid>
+                <Grid item xs={8}>
+                    <Typography gutterBottom
+                                className={classes.searchBarTitle}
+                                variant="h4">FIND YOUR MOVIE</Typography>
+                    <SearchBar handleInput={handleInput} handleSearchSubmit={handleSearchSubmit}></SearchBar>
```
Comment (dpryshchep) [inline src/header/Header.js:94]:
```
<SearchBar
    handleInput={handleInput}
    handleSearchSubmit={handleSearchSubmit}
/>
```

File: src/header/Header.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                </Grid>
+                <Grid item xs={8} className={classes.searchByBlock}>
+                    <ToggleGroup
+                        buttonGroupLabel="SEARCH BY"
+                        spacing={2}
+                        handleAlignment={handleSearchAlignment}
+                        buttons={searchBarState.buttons}/>
+                </Grid>
+            </Grid>
+            <ControlBar amountOfMovies={props.amountOfMovies} handleAlignment={handleSortAlignment} buttons={controlBarState.buttons}></ControlBar>
```
Comment (dpryshchep) [inline src/header/Header.js:104]:
```
<ControlBar
     amountOfMovies={props.amountOfMovies}
     handleAlignment={handleSortAlignment}
     buttons={controlBarState.buttons}
/>
```

**Comment by dpryshchep** [comment]
I would suggest to place store components in such way:
1. create a separate folder for each component (e.g. Button)
2. create a .jsx file for component itself in scope of Button folder
3. create a index.js file which will contain export of Button.jsx file
4. create a test files, mock, snapshots in scope of Button folder

In such way you will store all necessary files related to Button component in scope of one folder (tests, implementation, mock, etc.)

```
-- Button
   -- Button.jsx
   -- Button.test.js
   -- snapshots
   -- index.js
   -- etc.
```

File: src/header/Header.js (hunk 5 above)
Comment (mill9r) [reply src/header/Header.js]: @dpryshchep fixed

File: src/header/Header.js (hunk 4 above)
Comment (mill9r) [reply src/header/Header.js]: @dpryshchep fixed

File: src/common/ButtonGroup.js (hunk 2 above)
Comment (mill9r) [reply src/common/ButtonGroup.js]: @dpryshchep fixed

File: src/card/MovieCard.js (hunk 1 above)
Comment (mill9r) [reply src/card/MovieCard.js]: @dpryshchep fixed

**Pull request merged**